src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/CaseNoteEventPublisher.kt

```diff
@@ -5,31 +5,37 @@ import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.CaseNoteController
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.CaseNote
+import java.util.UUID
 
 enum class CaseNoteEventType {
   SENT,
 }
 
-class CaseNoteEvent(
+class CreateCaseNoteEvent(
   source: Any,
-  val type: CaseNoteEventType,
-  val caseNote: CaseNote,
+  val caseNoteId: UUID,
+  val sentBy: AuthUser,
   val detailUrl: String,
+  val referralId: UUID,
 ) : ApplicationEvent(source)
 
 @Component
 class CaseNoteEventPublisher(
   private val applicationEventPublisher: ApplicationEventPublisher,
   private val locationMapper: LocationMapper,
 ) {
+  // send the referralId as UUID since there's no guarantee that the session is still open
```

Comment:
True, but also true in a more general sense (i.e. "send primary key IDs instead of complex hydrated JPA entities ..."). Could we write a decision record about it, please, as part of this change?

It can go here: https://github.com/ministryofjustice/hmpps-interventions-service/tree/main/doc/adr

Reply:
I've added a new readme in doc/adr/0003-decouple-transactions-on-asynch-calls.md

let me know what you think

Reply:
(I have tested locally by building a docker container.)